integration-tests/velocity/src/main/java/org/apache/camel/quarkus/component/velocity/it/VelocityResource.java

```diff
@@ -0,0 +1,192 @@
+            endpointUrl = endpointUrl + "?contentCache=" + contentCache;
+        }
+        try {
+            final String response = producerTemplate.requestBodyAndHeaders(endpointUrl, message,
+                    headers,
+                    String.class);
+            LOG.infof("Got response from velocity: %s", response);
+            return Response
+                    .created(new URI("https://camel.apache.org/"))
+                    .entity(response)
+                    .build();
+        } catch (Exception e) {
+            if (exectFaiure != null && Boolean.parseBoolean(exectFaiure)) {
+                return Response
+                        .created(new URI("https://camel.apache.org/"))
+                        .entity(e.toString())
+                        .status(500)
+                        .build();
+            } else {
+                throw e;
+            }
+        }
+    }
+
+    @Path("/bodyAsDomainObject")
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response bodyAsDomainObject(Person person, @QueryParam("givenName") String givenName,
+            @QueryParam("familyName") String familyName) throws Exception {
+        LOG.infof("Sending to velocity: %s", person);
+        final String response = producerTemplate.requestBody("velocity://template/BodyAsDomainObject.vm", person,
+                String.class);
+        LOG.infof("Got response from velocity: %s", response);
+        return Response
+                .created(new URI("https://camel.apache.org/"))
+                .entity(response)
+                .build();
+    }
+
+    @Path("/templateViaHeader")
+    @POST
+    @Consumes(MediaType.TEXT_PLAIN)
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response templateViaHeader(String message, @QueryParam("body") String body, @QueryParam("item") String item,
+            @QueryParam("name") String name) throws Exception {
+        LOG.infof("Sending to velocity: %s", body);
+        Map<String, Object> headers = new HashMap() {
+            {
+                put("item", item);
+                put("name", name);
+                put(VelocityConstants.VELOCITY_TEMPLATE, message);
+            }
+        };
+        final String response = producerTemplate.requestBodyAndHeaders("velocity::dummy?allowTemplateFromHeader=true", body,
+                headers,
+                String.class);
+        LOG.infof("Got response from velocity: %s", response);
+        return Response
+                .created(new URI("https://camel.apache.org/"))
+                .entity(response)
+                .build();
+    }
+
+    @Path("/withProperties")
+    @POST
+    @Consumes(MediaType.TEXT_PLAIN)
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response withProperties(String message, @QueryParam("item") String item,
+            @QueryParam("name") String name) throws Exception {
+        LOG.infof("Sending to velocity: %s", message);
```

Comment:
Extensive logging slows down the test execution on machines with slow IO (which CI machines typically are). I think we should prefer debug level in situations like this. I should perhaps change the template that generates these messages. This is just a tip for the future.

Reply:
I've left logging here because it was generated by the command `fmvn cq:create -N -Dcq.artifactIdBase=foo-abc`. But in future I won't keep it